docs/docs/api-ref.md

```diff
@@ -5,257 +5,3008 @@ layout: docs
 
 <div class="alert alert-info">
     <b>Important:</b> More coming soon! This section is under active construction and might not reflect all the available functionality of the TSC library.
-    Until this reference is completed, we have noted the source files in the TSC library where you can get more information for individual endpoints.
-</div>
+    <!--- Until this reference is completed, we have noted the source files in the TSC library where you can get more information for individual endpoints. --->
+</div> 
+
+ 
+  
+The Tableau Server Client (TSC) is a Python library for the Tableau Server REST API. Using the TSC library, you can manage and change many of the Tableau Server and Tableau Online resources programmatically. You can use this library to create your own custom applications.
+
+The TSC API reference is organized by resource. The TSC library is modeled after the REST API. The methods, for example, `workbooks.get()`, correspond to the endpoints for resources, such as [workbooks](#workbooks), [users](#users), [views](#views), and [data sources](#data-sources). The model classes (for example, the [WorkbookItem class](#workbookitem-class) have attributes that represent the fields (`name`, `id`, `owner_id`) that are in the REST API request and response packages, or payloads. 
+
+|:---  |  
+| **Note:**  Some methods and features provided in the REST API might not be currently available in the TSC library.  In addition, the same limitations apply to the TSC library that apply to the REST API with respect to resources on Tableau Server and Tableau Online. For more information, see the [Tableau Server REST API Reference](http://onlinehelp.tableau.com/current/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#API_Reference%3FTocPath%3DAPI%2520Reference%7C_____0){:target="_blank"}.|
+
+ 
+
+* TOC  
+{:toc }
+<!-- {:toc ::options toc_levels="1,2"} -->
+
+<!-- {::toc_levels(1,2,3)} -->
+
+<br>
+<br>
 
-* TOC
-{:toc}
 
 ## Authentication
 
-Source files: server/endpoint/auth_endpoint.py, models/tableau_auth.py
+You can use the TSC library to manage authentication, so you can sign in and sign out of Tableau Server and Tableau Online. The authentication resources for Tableau Server are defined in the `TableauAuth` class and they correspond to the authentication attributes you can access using the Tableau Server REST API. 
```

Comment:
No problem. It took me a while to get back on this.  I just made another commit and added the update_connections for workbooks. That is what caused the merge conflict.